Common/src/geometry_structure.cpp

```diff
@@ -15564,49 +15564,349 @@ void CPhysicalGeometry::SetColorGrid_Parallel(CConfig *config) {
+    
+    su2double Area = 0.0;
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      Area += Normal[iDim]*Normal[iDim];
+    Area = sqrt(Area);
+    
+    /*--- Aspect ratio is the ratio between the largest and smallest
+     faces making up the boundary of the dual CV and is a
+     measure of the aspect ratio of the dual control volume. Smaller
+     is better (closer to isotropic). ----*/
+    
+    if (node[iPoint]->GetDomain()) {
+      if (Area < Area_Min[iPoint]) Area_Min[iPoint] = Area;
+      if (Area > Area_Max[iPoint]) Area_Max[iPoint] = Area;
+    }
+    
+    if (node[jPoint]->GetDomain()) {
+      if (Area < Area_Min[jPoint]) Area_Min[jPoint] = Area;
+      if (Area > Area_Max[jPoint]) Area_Max[jPoint] = Area;
+    }
+    
+    if (Area == 0.0) {
+      char buf[200];
+      SPRINTF(buf, "Zero-area CV face found for edge (%lu,%lu).",
+              GlobalIndex_i, GlobalIndex_j);
+      SU2_MPI::Error(string(buf), CURRENT_FUNCTION);
+    }
+    
+    /*-- Unit normal for the current edge. Recall that this normal
+     is computed by summing the normals or adjacent faces along
+     the edge between iPoint & jPoint. ---*/
+    
+    vector<su2double> UnitNormal(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      UnitNormal[iDim] = Normal[iDim]/Area;
+    
+    /*--- Get the coordinates for point i & j. ---*/
+    
+    const su2double *Coord_i = node[iPoint]->GetCoord();
+    const su2double *Coord_j = node[jPoint]->GetCoord();
+    
+    /*--- Compute vector pointing from iPoint to jPoint and the distance. ---*/
+    
+    su2double dist_ij = 0.0;
+    vector<su2double> Edge_Vector(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      Edge_Vector[iDim] = Coord_j[iDim]-Coord_i[iDim];
+      dist_ij += Edge_Vector[iDim]*Edge_Vector[iDim];
+    }
+    dist_ij = sqrt(dist_ij);
+    
+    /*-- Normalize the edge vector. ---*/
+    
+    vector<su2double> Unit_Edge_Vector(nDim);
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      Unit_Edge_Vector[iDim] = Edge_Vector[iDim]/dist_ij;
+    
+    /*--- Compute the angle between the unit normal associated
+     with the edge and the unit vector pointing from iPoint to jPoint. ---*/
+    
+    su2double dot_prod = 0.0;
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      dot_prod += UnitNormal[iDim]*Unit_Edge_Vector[iDim];
+    
+    /*--- The definition of orthogonality is an area-weighted average of
+     90 degrees minus the angle between the face area unit normal and
+     the vector between i & j. If the two are perfectly aligned, then
+     the orthogonality is the desired max of 90 degrees. If they are
+     not aligned, the orthogonality will reduce from there. Good values
+     are close to 90 degress, poor values are typically below 20 degress. ---*/
+    
+    if (node[iPoint]->GetDomain()) {
+      Orthogonality[iPoint] += Area*(90.0 - acos(dot_prod)*180.0/PI_NUMBER);
```

Comment:
Should this be computed such that it is always in range 0-90? (135 is as bad as 45)

Reply:
During creation of the dual CVs, we orient the sub-face area of a CV in the direction from iPoint to jPoint. So when we take the dot product with the position vector from iPoint to jPoint, the max value will be 90 deg by construction (giving us a 0.0 orthogonality value - worst). If they are perfectly aligned, the dot product is zero (giving us a 90.0 orthogonality angle - best).